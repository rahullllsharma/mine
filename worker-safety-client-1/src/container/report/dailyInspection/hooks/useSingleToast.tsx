import type { ToastType } from "@/components/shared/toast/toastItem/Toast";

import type { ToastItem } from "@/components/shared/toast/context/ToastContext";
import { useContext } from "react";
import ToastContext from "@/components/shared/toast/context/ToastContext";

type ReplaceActiveToast = (type: ToastType, message: string) => void;

/**
 * Only display one toast at each time.
 *
 * @description
 * Pop the last toast and replace with a new one.
 *
 * @returns [(type: ToastType, message: string) => void]
 *
 * @deprecated DO NOT USE THIS, will be removed
 */
const useSingleToast = (): [ReplaceActiveToast] => {
  let lastActiveToast: ToastItem;
  const toastCtx = useContext(ToastContext);

  const replaceActiveToast: ReplaceActiveToast = (type, message) => {
    if (!lastActiveToast) {
      [lastActiveToast] = toastCtx?.items || [];
    }

    if (toastCtx?.items.some(item => item.id === lastActiveToast?.id)) {
      toastCtx?.dismissToast(lastActiveToast.id);
    }

    toastCtx?.pushToast(type, message);
    [lastActiveToast] = toastCtx?.items || [];
  };

  return [replaceActiveToast];
};

export default useSingleToast;
