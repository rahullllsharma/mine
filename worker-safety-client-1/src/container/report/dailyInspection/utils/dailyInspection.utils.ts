import type { DailyInspectionReportForms } from "../types";
import type { DailyReport } from "@/types/report/DailyReport";
import type { DailyReportRecommendations } from "@/types/report/DailyReportRecommendations";
import { convertDateToString } from "@/utils/date/helper";
import { workScheduleFormInputPrefix } from "../../workSchedule/WorkSchedule";

// Types
type DailyReportResponse = Omit<DailyReport, "sections"> & {
  sections: Required<
    {
      [k in keyof Required<DailyReport>["sections"]]:
        | Required<DailyReport>["sections"][k]
        | null;
    }
  >;
};

type ApplyRecommendationsFn = (params: {
  recommendations: DailyReportRecommendations["dailyReport"] | null;
  dailyReport: DailyReportResponse | undefined;
}) => DailyReport | undefined;

type ApplyDefaultsToArrayFieldsFn = (
  params: DailyReport | undefined
) => DailyReport | undefined;

// Private functions

const getCrewSectionRecommendations = ({
  crew,
}: Pick<DailyReportRecommendations["dailyReport"], "crew">) =>
  crew === null
    ? null
    : {
        foremanName: crew.foremanName,
        contractor: crew.constructionCompany,
      };

const getSafetyAndComplianceSectionRecommendations = ({
  safetyAndCompliance,
}: Pick<DailyReportRecommendations["dailyReport"], "safetyAndCompliance">) =>
  safetyAndCompliance === null
    ? null
    : {
        plans: {
          comprehensivePHAConducted: safetyAndCompliance.phaCompletion,
        },
        systemOperatingProcedures: {
          sopId: safetyAndCompliance.sopNumber,
          sopType: safetyAndCompliance.sopType,
          sopStepsCalledIn: safetyAndCompliance.stepsCalledIn,
        },
      };

// Public functions

function getWorkScheduleStartDateTime(
  formData: DailyInspectionReportForms
): string {
  const startDateTime = formData?.[workScheduleFormInputPrefix]?.startDatetime;
  if (startDateTime) {
    return convertDateToString(startDateTime);
  }

  throw new Error(
    "Unable to get start date time from Work Schedule section in form"
  );
}

function getStartDateTimeFromSavedReport(dailyReport?: DailyReport): string {
  const startDateTime = dailyReport?.sections?.workSchedule?.startDatetime;
  if (startDateTime) {
    return convertDateToString(startDateTime);
  }

  throw new Error("Unable to get start date time from existing daily report");
}

/** Generate the daily report based on the recommendations */
const applyRecommendationsToSections: ApplyRecommendationsFn = ({
  recommendations,
  dailyReport,
}) => {
  if (!recommendations) {
    return dailyReport as DailyReport;
  }

  const { crew, safetyAndCompliance } = dailyReport?.sections || {};

  // All sections filled
  if (!!crew && !!safetyAndCompliance) {
    return dailyReport as DailyReport;
  }

  return {
    ...dailyReport,
    sections: {
      ...(dailyReport?.sections || {}),
      crew: crew || getCrewSectionRecommendations(recommendations),
      safetyAndCompliance:
        safetyAndCompliance ||
        getSafetyAndComplianceSectionRecommendations(recommendations),
    },
  } as DailyReport;
};

/** Apply an empty array for all attachment fields (documents and photos) for all sections */
const applyDefaultsToArrayFields: ApplyDefaultsToArrayFieldsFn =
  dailyReport => {
    if (!dailyReport || !dailyReport?.sections) {
      return dailyReport;
    }

    const modifiedDailyReport = {
      ...dailyReport,
      sections: {
        ...dailyReport.sections,
      },
    };

    // we could this recursively or use lodash but for now, we only have these scenarios.
    if (modifiedDailyReport.sections?.crew) {
      modifiedDailyReport.sections.crew.documents ??= [];
    }

    if (modifiedDailyReport.sections?.attachments) {
      modifiedDailyReport.sections.attachments.documents ??= [];
      modifiedDailyReport.sections.attachments.photos ??= [];
    }

    return modifiedDailyReport;
  };

export type { DailyReportResponse };

export {
  applyRecommendationsToSections,
  applyDefaultsToArrayFields,
  getWorkScheduleStartDateTime,
  getStartDateTimeFromSavedReport,
};
