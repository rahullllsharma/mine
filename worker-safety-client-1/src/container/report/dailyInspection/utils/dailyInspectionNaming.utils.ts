import { useTenantStore } from "@/store/tenant/useTenantStore.store";

const getActivityTaskName = () => {
  const { activity, task } = useTenantStore.getState().getAllEntities();
  return `${activity.label} ${task.labelPlural}`;
};

export { getActivityTaskName };
