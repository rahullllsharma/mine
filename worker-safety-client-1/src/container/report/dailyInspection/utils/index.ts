export * from "./shallowReplaceUrl.utils";
export * from "./preserveDailyReportSections.utils";
export * from "./markDailyReportSectionsCompletedFromReport.utils";
export * from "./dailyInspection.utils";
export * from "./dailyInspectionNaming.utils";
