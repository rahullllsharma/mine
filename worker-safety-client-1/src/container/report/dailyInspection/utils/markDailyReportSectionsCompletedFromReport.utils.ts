import type { DailyReport as DailyInspectionReportType } from "@/types/report/DailyReport";
import type { DailyReportInputs } from "@/types/report/DailyReportInputs";
import type { DailyInspectionReportMultiStep } from "../types";

type DailyReportInputsKeys = keyof NonNullable<DailyReportInputs>;

/**
 * Checks if the current section is completed or not.
 * It should work for both new (with sectionIsValid) and old sections (without the flag)
 * For new reports, `sectionIsValid` can assume 3 values (true | false | null)
 *
 * When sectionIsValid = NULL, it should fallback to the old way of checking
 * if a section is valid or not, which is basically checking if the section has values.
 */
const isSectionCompleted = (
  section:
    | NonNullable<DailyInspectionReportType["sections"]>[DailyReportInputsKeys]
    | null
    | undefined = {}
) => {
  if (
    section !== null &&
    Object.prototype.hasOwnProperty.call(section, "sectionIsValid")
  ) {
    return section?.sectionIsValid;
  }

  return Object.values(section || {}).length !== 0;
};

function getDailyReportMetadataWithCompletedSections(
  metadata: DailyInspectionReportMultiStep[],
  savedDailyReport?: DailyInspectionReportType
): DailyInspectionReportMultiStep[] {
  if (!savedDailyReport?.sections) {
    return metadata;
  }

  return metadata.map(item => {
    const hasSectionCompleted = isSectionCompleted(
      savedDailyReport.sections?.[item.id as DailyReportInputsKeys]
    );

    return {
      ...item,
      status: hasSectionCompleted ? "completed" : item?.status,
    };
  });
}

export { getDailyReportMetadataWithCompletedSections };
