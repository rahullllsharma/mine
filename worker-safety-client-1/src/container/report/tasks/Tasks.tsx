import type { TaskHazardAggregator } from "@/types/project/HazardAggregator";
import type { FormEventHandler } from "react";

import cx from "classnames";
import { Controller, useFormContext } from "react-hook-form";

import TaskHeader from "@/components/layout/taskCard/header/TaskHeader";
import TaskCard from "@/components/layout/taskCard/TaskCard";
import Checkbox from "@/components/shared/checkbox/Checkbox";
import { getBorderColorByRiskLevel } from "@/utils/risk";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";

type TaskCardListItem = TaskHazardAggregator & {
  isSelected: boolean;
};

export type TasksProps = {
  tasks: TaskCardListItem[];
  isCompleted?: boolean;
};

export const taskFormInputPrefix = "taskSelection";

export default function Tasks({
  tasks = [],
  isCompleted,
}: TasksProps): JSX.Element {
  const {
    workPackage,
    location,
    task: taskEntity,
    activity: activityEntity,
  } = useTenantStore(state => state.getAllEntities());

  const isTaskVisible = (isSelected: boolean): boolean => {
    return !isCompleted || isSelected;
  };

  const filteredTasks =
    tasks?.length > 0
      ? tasks.filter(task => isTaskVisible(task.isSelected))
      : [];

  const hasTasks = filteredTasks?.length > 0;

  const { watch, getValues, setValue } = useFormContext();

  const fieldsAsArray = Object.values(watch()?.[taskFormInputPrefix] || []);

  // This is only needed if no selection is done
  const areAllTasksSelected =
    hasTasks &&
    (fieldsAsArray.length === 0
      ? filteredTasks.every(({ isSelected }) => !!isSelected)
      : fieldsAsArray.every(value => !!value));

  const onToggleAllHandler: FormEventHandler<HTMLInputElement> = ({
    currentTarget: { checked },
  }) => {
    Object.keys(getValues()[taskFormInputPrefix]).forEach(task => {
      setValue(`${taskFormInputPrefix}.${task}`, checked, {
        shouldDirty: true,
      });
    });
  };

  const onToggleHandler = (id: string) => {
    const inputName = `${taskFormInputPrefix}.${id}`;
    const inputValue = getValues(inputName);
    setValue(inputName, !inputValue, { shouldDirty: true });
  };

  return (
    <>
      <h3 className="text-xl text-neutral-shade-100 font-semibold">
        {`${activityEntity.label} ${taskEntity.label.toLowerCase()} selection`}
      </h3>
      <p className="text-sm text-neutral-shade-100">
        {`Select the ${taskEntity.labelPlural.toLowerCase()} you were responsible for overseeing at this ${location.label.toLowerCase()}`}
        .
      </p>
      {hasTasks ? (
        <div className="mt-4">
          {!isCompleted && (
            <label className="flex items-center mb-4 gap-4" htmlFor="all">
              <Checkbox
                id="all"
                checked={areAllTasksSelected}
                onChange={onToggleAllHandler}
              />
              <span className="font-semibold text-neutral-shade-75 hover:cursor-pointer">
                {`All ${taskEntity.labelPlural.toLowerCase()}`}
              </span>
            </label>
          )}
          <ul className="mt-4">
            {filteredTasks.map(
              ({ id, riskLevel, name, isSelected, activity }) => (
                <li key={id} className="flex gap-4 items-center mb-2">
                  {/* if we start to have performance issues, we should consider only using uncontrolled inputs */}
                  {!isCompleted && (
                    <Controller
                      name={`${taskFormInputPrefix}.${id}`}
                      defaultValue={isSelected}
                      render={({ field }) => (
                        <Checkbox
                          {...field}
                          checked={field.value}
                          value={`${taskFormInputPrefix}.${id}`}
                          aria-label={name}
                          className="mr-1"
                        />
                      )}
                    />
                  )}
                  <TaskCard
                    className={cx(
                      "flex-1 mb-0",
                      getBorderColorByRiskLevel(riskLevel)
                    )}
                    taskHeader={
                      <TaskHeader
                        headerText={name}
                        subHeaderText={activity?.name}
                        riskLevel={riskLevel}
                        onClick={() => onToggleHandler(id)}
                      />
                    }
                  />
                </li>
              )
            )}
          </ul>
        </div>
      ) : (
        <p className="mt-8 ">
          {`There are currently no ${activityEntity.labelPlural.toLowerCase()} in progress between the set
           ${workPackage.attributes.primeContractor.label.toLowerCase()} start and end dates.`}
        </p>
      )}
    </>
  );
}
