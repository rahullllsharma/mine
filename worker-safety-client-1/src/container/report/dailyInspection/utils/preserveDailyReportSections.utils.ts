import type { DailyReport } from "@/types/report/DailyReport";

import { jobHazardAnalysisFormInputPrefix } from "@/components/report/jobReport/JobReportCard";
import { taskFormInputPrefix } from "../../tasks/Tasks";
import { workScheduleFormInputPrefix } from "../../workSchedule/WorkSchedule";

type DailyReportSectionsKeys = keyof NonNullable<DailyReport["sections"]>;

type DailyInspectionReportResettableSections = Extract<
  DailyReportSectionsKeys,
  typeof workScheduleFormInputPrefix | typeof taskFormInputPrefix
>;

const revalidateSectionsBySection: {
  [k in DailyInspectionReportResettableSections]: Array<
    Partial<DailyReportSectionsKeys>
  >;
} = {
  [workScheduleFormInputPrefix]: [
    taskFormInputPrefix,
    jobHazardAnalysisFormInputPrefix,
  ],
  [taskFormInputPrefix]: [jobHazardAnalysisFormInputPrefix],
};

function preserveDailyReportSections(
  currentSection: DailyInspectionReportResettableSections,
  dailyReport?: DailyReport
): Partial<NonNullable<DailyReport["sections"]>> {
  const staledSectionsBasedOnCurrentSection =
    revalidateSectionsBySection[currentSection] || [];

  const {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    sections: { [currentSection]: _, ...remaining } = {},
  } = dailyReport || {};

  return Object.fromEntries(
    Object.entries(remaining).filter(
      ([section]: Array<DailyReportSectionsKeys>) =>
        !staledSectionsBasedOnCurrentSection.includes(section)
    )
  );
}

function defaultSectionIsValidPropBySections(
  dailyReportPayload: Required<Pick<DailyReport, "sections">>["sections"]
) {
  return Object.fromEntries(
    Object.entries(dailyReportPayload).map(([key, value]) => [
      key,
      value === null
        ? null
        : {
            ...value,
            ...{ sectionIsValid: value?.sectionIsValid ?? null },
          },
    ])
  );
}

// export types
export type { DailyInspectionReportResettableSections };

// exports
export {
  revalidateSectionsBySection,
  preserveDailyReportSections,
  defaultSectionIsValidPropBySections,
};
