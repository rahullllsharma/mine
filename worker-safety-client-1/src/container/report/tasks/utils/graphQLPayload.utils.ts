import type { TaskSelectionInputs } from "@/types/report/DailyReportInputs";
import type { Location } from "@/types/project/Location";
import type {
  HazardAggregator,
  TaskHazardAggregator,
} from "@/types/project/HazardAggregator";
import { taskFormInputPrefix } from "../Tasks";

export type TasksGraphQLPayloadParams = {
  [taskFormInputPrefix]?: Record<string, true>;
};

export type TasksGraphQLPayloadOptions = { location: Location };

type ReturnedTasksGraphQLPayload = {
  taskSelection: {
    selectedTasks: TaskSelectionInputs[];
  };
};

const filterAndTransformSelectedTasks =
  (taskSelection: TasksGraphQLPayloadParams["taskSelection"]) =>
  // TODO: this should be rewritten to include either the SiteConditionHazardAggregator Or TaskHazardAggregator
  (
    acc: TaskSelectionInputs[],
    { id, name, riskLevel }: HazardAggregator | TaskHazardAggregator
  ) => {
    if (taskSelection?.[id]) {
      acc.push({
        id,
        name,
        riskLevel,
      });
    }

    return acc;
  };

export function transformGraphQLPayload(
  formData: TasksGraphQLPayloadParams,
  options: TasksGraphQLPayloadOptions
): ReturnedTasksGraphQLPayload {
  const { taskSelection } = formData || {};
  const selectedTasks = options.location.tasks.reduce(
    filterAndTransformSelectedTasks(taskSelection),
    [] as TaskSelectionInputs[]
  );

  return {
    [taskFormInputPrefix]: {
      selectedTasks,
    },
  };
}
