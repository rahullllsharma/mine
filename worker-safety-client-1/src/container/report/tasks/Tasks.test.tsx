import type { TasksProps } from "./Tasks";
import { screen, fireEvent, within } from "@testing-library/react";

import { useProjects } from "@/hooks/useProjects";
import { transformTasksToListTasks } from "@/utils/task";

import { formRender, mockTenantStore } from "@/utils/dev/jest";
import Tasks from "./Tasks";

// eslint-disable-next-line react-hooks/rules-of-hooks
const { tasks } = useProjects()[0].locations[0];
const projectTasks = transformTasksToListTasks(tasks);

describe(Tasks.name, () => {
  mockTenantStore();
  it("should render the titles", () => {
    const { asFragment } = formRender(<Tasks tasks={projectTasks} />);

    screen.getByRole("heading", { level: 3 });
    screen.getByText(
      "Select the tasks you were responsible for overseeing at this location."
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with the default selections", () => {
    formRender(
      <Tasks
        tasks={projectTasks.map(task => ({
          ...task,
          isSelected: false,
        }))}
      />
    );

    screen.getAllByRole("listitem").forEach(item => {
      expect(within(item).getByRole("checkbox")).not.toBeChecked();
    });
  });

  it.each([false, undefined, null, []] as TasksProps["tasks"][])(
    "should render the empty state when no tasks are found",
    taskListItem => {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      formRender(<Tasks tasks={taskListItem} />);
      screen.getByText(
        "There are currently no activities in progress between the set contractor start and end dates."
      );
    }
  );

  describe("when has tasks", () => {
    it("should have the all tasks checkbox checked if all tasks are selected", () => {
      const allSelectedTasks = transformTasksToListTasks(tasks, tasks);
      formRender(<Tasks tasks={allSelectedTasks} />);

      expect(
        screen.getByRole("checkbox", {
          name: /all tasks/i,
        })
      ).toBeChecked();
    });

    it("should have the all tasks checkbox unchecked if one of tasks is NOT selected", () => {
      const projectsWithUnselected = projectTasks.concat();
      projectsWithUnselected[0].isSelected = false;

      formRender(<Tasks tasks={projectsWithUnselected} />);

      expect(
        screen.getByRole("checkbox", {
          name: /all tasks/i,
        })
      ).not.toBeChecked();
    });

    describe("for each tasks", () => {
      it("should have be checked/unchecked based if it is selected or not", () => {
        const projectsWithUnselected = projectTasks.concat();
        projectsWithUnselected[0].isSelected = false;

        formRender(<Tasks tasks={projectsWithUnselected} />);

        projectsWithUnselected.forEach(task => {
          expect(
            (
              screen.getByRole("checkbox", {
                name: task.name,
              }) as HTMLInputElement
            ).checked
          ).toBe(task.isSelected);
        });
      });
    });

    describe("when clicking the 'All Tasks' checkbox", () => {
      it.each(["on", "off"])(
        "and the checkbox is %s, should change all task checkboxes with the opposite value",
        value => {
          const isSelected = value === "on";
          formRender(
            <Tasks
              tasks={projectTasks.map(task => ({ ...task, isSelected }))}
            />
          );

          fireEvent.click(screen.getByRole("checkbox", { name: /all tasks/i }));

          // since we trigger the All Tasks checkbox,
          // we query if NO checkbox has the initial value
          expect(
            screen.queryAllByRole("checkbox", { checked: isSelected })
          ).toHaveLength(0);
        }
      );
    });

    describe("when clicking a task", () => {
      it.each(["on", "off"])(
        "and the task is %s, should change that task with the opposite value",
        value => {
          const isSelected = value === "on";

          const selected = projectTasks[1];
          const taskListItem = projectTasks.map(task => ({
            ...task,
            isSelected: task.id === selected.id ? isSelected : task.isSelected,
          }));

          formRender(<Tasks tasks={taskListItem} />);

          fireEvent.click(
            screen.getByRole("checkbox", { name: selected.name })
          );

          screen.queryByRole("checkbox", {
            name: selected.name,
            checked: !isSelected,
          });
        }
      );
    });
  });
});
