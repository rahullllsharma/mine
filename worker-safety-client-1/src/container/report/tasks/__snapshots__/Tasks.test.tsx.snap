// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks should render the titles 1`] = `
<DocumentFragment>
  <h3
    class="text-xl text-neutral-shade-100 font-semibold"
  >
    Activity task selection
  </h3>
  <p
    class="text-sm text-neutral-shade-100"
  >
    Select the tasks you were responsible for overseeing at this location.
  </p>
  <div
    class="mt-4"
  >
    <label
      class="flex items-center mb-4 gap-4"
      for="all"
    >
      <input
        class="flex-shrink-0 root"
        id="all"
        type="checkbox"
      />
      <span
        class="font-semibold text-neutral-shade-75 hover:cursor-pointer"
      >
        All tasks
      </span>
    </label>
    <ul
      class="mt-4"
    >
      <li
        class="flex gap-4 items-center mb-2"
      >
        <input
          aria-label="Loading/Unloading"
          class="flex-shrink-0 root mr-1"
          name="taskSelection.task1"
          type="checkbox"
          value="taskSelection.task1"
        />
        <div
          class="bg-white shadow-10 rounded-lg mb-2.5 border-l-8 whitespace-break-spaces flex-1 mb-0 border-risk-medium"
          data-testid="taskCard"
        >
          <header
            class="flex justify-between items-center"
          >
            <div
              class="flex flex-1 items-stretch h-auto p-3"
            >
              <div
                class="flex flex-1 flex-wrap items-stretch gap-y-3 gap-x-6 md:gap-0 md:flex-row md:flex-nowrap md:items-center mr-6 outline-none"
                role="button"
                tabindex="0"
              >
                <div
                  class="text-left text-base text-neutral-shade-100 font-bold flex flex-auto m-0 items-center gap-3 w-full md:flex-initial md:gap-4 md:w-auto md:ml-3 md:mr-4"
                >
                  <div>
                    <span
                      class="text-action-label font-action-label text-neutrals-primary font-semibold"
                    >
                      Loading/Unloading
                    </span>
                    <span
                      class=" pl-1 text-tiny font-medium text-neutral-shade-58"
                    >
                      activity1
                    </span>
                  </div>
                </div>
                <span
                  class="inline-flex items-center"
                >
                  <span
                    class="inline-flex items-center w-full"
                  >
                    <div
                      class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-medium text-neutral-shade-100"
                      role="note"
                    >
                      <span
                        class="flex items-center px-1"
                      >
                        <i
                          aria-hidden="true"
                          class=""
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M9.74 11.921c-.497-.053-.91.09-1.21.441-.364.428-.53.811-.53 1.138H6c0-.998.492-1.83 1.008-2.435.796-.934 1.902-1.245 2.947-1.132 1.021.11 2.01.619 2.752 1.36.453.453 1.03.73 1.552.786.498.053.91-.09 1.21-.441.365-.428.531-.812.531-1.138h2c0 .998-.492 1.83-1.008 2.435-.796.934-1.902 1.245-2.947 1.132-1.021-.11-2.01-.619-2.752-1.36-.453-.453-1.03-.73-1.552-.786z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        <span
                          class="text-tiny mr-1"
                        >
                          MEDIUM
                        </span>
                      </span>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </header>
        </div>
      </li>
      <li
        class="flex gap-4 items-center mb-2"
      >
        <input
          aria-label="Pipe Face Alignment"
          class="flex-shrink-0 root mr-1"
          name="taskSelection.task2"
          type="checkbox"
          value="taskSelection.task2"
        />
        <div
          class="bg-white shadow-10 rounded-lg mb-2.5 border-l-8 whitespace-break-spaces flex-1 mb-0 border-risk-medium"
          data-testid="taskCard"
        >
          <header
            class="flex justify-between items-center"
          >
            <div
              class="flex flex-1 items-stretch h-auto p-3"
            >
              <div
                class="flex flex-1 flex-wrap items-stretch gap-y-3 gap-x-6 md:gap-0 md:flex-row md:flex-nowrap md:items-center mr-6 outline-none"
                role="button"
                tabindex="0"
              >
                <div
                  class="text-left text-base text-neutral-shade-100 font-bold flex flex-auto m-0 items-center gap-3 w-full md:flex-initial md:gap-4 md:w-auto md:ml-3 md:mr-4"
                >
                  <div>
                    <span
                      class="text-action-label font-action-label text-neutrals-primary font-semibold"
                    >
                      Pipe Face Alignment
                    </span>
                    <span
                      class=" pl-1 text-tiny font-medium text-neutral-shade-58"
                    >
                      activity1
                    </span>
                  </div>
                </div>
                <span
                  class="inline-flex items-center"
                >
                  <span
                    class="inline-flex items-center w-full"
                  >
                    <div
                      class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-medium text-neutral-shade-100"
                      role="note"
                    >
                      <span
                        class="flex items-center px-1"
                      >
                        <i
                          aria-hidden="true"
                          class=""
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M9.74 11.921c-.497-.053-.91.09-1.21.441-.364.428-.53.811-.53 1.138H6c0-.998.492-1.83 1.008-2.435.796-.934 1.902-1.245 2.947-1.132 1.021.11 2.01.619 2.752 1.36.453.453 1.03.73 1.552.786.498.053.91-.09 1.21-.441.365-.428.531-.812.531-1.138h2c0 .998-.492 1.83-1.008 2.435-.796.934-1.902 1.245-2.947 1.132-1.021-.11-2.01-.619-2.752-1.36-.453-.453-1.03-.73-1.552-.786z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        <span
                          class="text-tiny mr-1"
                        >
                          MEDIUM
                        </span>
                      </span>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </header>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
